packages/backend/src/tokens/keys.ts

```diff
@@ -151,13 +142,21 @@ export async function loadClerkJWKFromRemote({
       .sort()
       .join(', ');
 
-    throw new TokenVerificationError({
-      action: TokenVerificationErrorAction.ContactSupport,
-      message: `Unable to find a signing key in JWKS that matches the kid='${kid}' of the provided session token. Please make sure that the __session cookie or the HTTP authorization header contain a Clerk-generated session JWT.${
-        jwkKeys ? ` The following kid is available: ${jwkKeys}` : ''
-      }`,
-      reason: TokenVerificationErrorReason.RemoteJWKMissing,
-    });
+    if (!jwkKeys) {
+      throw new TokenVerificationError({
+        action: TokenVerificationErrorAction.ContactSupport,
+        message: `No signing key found in JWKS. This is likely a bug in Clerk. (cache=${JSON.stringify(
+          cache,
+        )}, lastUpdatedAt=${lastUpdatedAt}, kid=${kid})`,
+        reason: TokenVerificationErrorReason.RemoteJWKMissing,
+      });
```

Comment:
Why do we need this ? Are we not confident in the fix, or is this another known complex issue ?

Reply:
Oops, I think this logic is wrong. The thought is that the cache should never be empty at this point, so it helps distinguish between failure states. It shouldn't be possible though.

I feel confident in the fix though, happy to remove.